Multisig/UI/Settings/OwnerKeyManagement/KeystoneOwnerKey/KeystoneSignFlow.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  KeystoneSignFlow.swift
+//  Multisig
+//
+//  Created by Zhiying Fan on 6/9/2022.
+//  Copyright © 2022 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+import URRegistry
+
+final class KeystoneSignFlow: UIFlow {
+    var signCompletion: ((_ unmarshaledSignature: SECP256K1.UnmarshaledSignature) -> Void)?
+    
+    private let signInfo: KeystoneSignInfo
+    private let signRequest: KeystoneSignRequest
+    
+    init?(signInfo: KeystoneSignInfo, completion: @escaping (Bool) -> Void) {
+        guard let signRequest = signInfo.signRequest else { return nil }
+        self.signInfo = signInfo
+        self.signRequest = signRequest
+        super.init(completion: completion)
+    }
+    
+    override func start() {
+        requestSignature()
+    }
+    
+    private func requestSignature() {
+        URRegistry.shared.setSignRequestUREncoder(with: signRequest)
+        let signVC = UIHostingController(rootView: KeystoneRequestSignatureView(onTap: { [weak self] in
+            self?.presentScanner()
+        }))
+        signVC.navigationItem.title = "Request signature"
+        
+        let ribbon = ViewControllerFactory.ribbonWith(viewController: signVC)
+        ribbon.storedChain = signInfo.chain
+        
+        show(ribbon)
+    }
+    
+    private func presentScanner() {
+        let vc = QRCodeScannerViewController()
```

Comment:
should be moved to the factory, seems to be similar to the one used in ConnectKeystoneFlow